### DISCUSSION TITLE
Fully Customizing error msg [Question]

### DISCUSSION THREAD

**Discussion by kr1tx** [dashingsoft/pyarmor#1131] [Q&A] [answered]
is it possible to customize this message? I dont mean the **License has expired** string, i mean the entire error message.
```
RuntimeError: License has expired (1:10937)
[284] Failed to execute script 'Program' due to unhandled exception!
```

**Comment by jondy** [comment]
Please read the 8.0 documentation.

**Reply by kr1tx** [reply]
Already read it but `pyarmor cfg on_error=2` doesn't work! *Already bought the pro version*

**Comment by jondy** [comment, marked as answer]
Sorry, I don't catch your question is `entire message`. 

This message is reported by PyInstaller, it's out of pyarmor. You can consult PyInstaller loader documentation to find if it's possible.

`on_error` only works for execute the obfuscated scripts by Python interpreter directly. If the script is loaded by PyInstaller loader, it may not work as expected.

**Reply by kr1tx** [reply]
The following error is only from pyarmor i did not use pyinstaller. Please allow me to explain more, I created a file called `messages.cfg` within the .pyarmor folder and added the following into it *as I read from the pyarmor 8.0 doc.
```
[runtime.message]
error_1 = ErrorMSG: License has expired
```
So the problem is that I don't want this ugly message to be shown when the license is expired, instead i want only `ErrorMSG: License has expired` to be shown.
```
Traceback (most recent call last):
  File "dist\Script.py", line 2, in <module>
    from pyarmor_runtime_xxxxxx import __pyarmor__
  File "dist\pyarmor_runtime_xxxxxx\__init__.py", line 2, in <module>
    from .pyarmor_runtime import __pyarmor__
RuntimeError: ErrorMSG: License has expired (1:10937)
```

**Reply by jondy** [reply]
So do you `pyarmor cfg on_error=1`?

**Reply by kr1tx** [reply]
Yes I have used it and it didnt work also `pyarmor cfg on_error=2`

**Reply by jondy** [reply]
I can't reproduce it.

Could you report an issue with steps to reproduce? Please provide necessary informations as describe in the but template.

**Reply by kr1tx** [reply]
Done
